src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -262,34 +260,23 @@ def list_role_assignments(cmd,  # pylint: disable=too-many-locals, too-many-bran
 
     # Fill in role definition names
     if fill_role_definition_name:
-        worker = MultiAPIAdaptor(cmd.cli_ctx)
         role_defs = list(definitions_client.list(
             scope=scope or ('/subscriptions/' + definitions_client._config.subscription_id)))
-        role_dics = {i.id: worker.get_role_property(i, 'role_name') for i in role_defs}
-        for i in results:
-            if not i.get('roleDefinitionName'):
-                if role_dics.get(worker.get_role_property(i, 'roleDefinitionId')):
```

Comment:
A `RoleAssignment` object must have `roleDefinitionId` property: https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/list-for-scope, so this statement is also unnecessary.